src/main/resources/META-INF/resources/primefaces/datatable/datatable.js

```diff
@@ -2399,6 +2399,12 @@ PrimeFaces.widget.DataTable = PrimeFaces.widget.DeferredWidget.extend({
         else {
             this.showRowEditors(row);
 
+            let inputs=$(row).find('input, select, textarea, button');
+            if (inputs.length > 0) {
+                let firstInput = inputs.first();
+                firstInput.focus();
```

Comment:
I think you should just make this one line with `inputs.first().focus();` no need to declare variable for this yet.